src/api/java/com/minecolonies/api/colony/requestsystem/requestable/IConcreteDeliverable.java

```diff
@@ -0,0 +1,17 @@
+package com.minecolonies.api.colony.requestsystem.requestable;
+
+import net.minecraft.item.ItemStack;
+
+import java.util.List;
+
+/**
+ * An {@link IConcreteDeliverable} is an Requestable that can be looked up fast in the warehouse, and delivered.
+ */
+public interface IConcreteDeliverable extends IDeliverable
```

Comment:
Let's discuss this. I started to do that and it felt inside out, because we're comparing two collections to see if there is any overlap. The deliverable list is likely the smallest of those, and thus needs to be the one we iterate over. I'm guessing the matches you're proposing would be to send in a list, but that would be the full inventory and negate the gains we're looking for.